Content.Server/Nyanotrasen/Abilities/Psionics/Abilities/TelegnosisPowerSystem.cs

```diff
@@ -53,15 +62,57 @@ private void OnShutdown(EntityUid uid, TelegnosisPowerComponent component, Compo
         private void OnPowerUsed(EntityUid uid, TelegnosisPowerComponent component, TelegnosisPowerActionEvent args)
         {
             var projection = Spawn(component.Prototype, Transform(uid).Coordinates);
-            Transform(projection).AttachToGridOrMap();
+
+            _mind.ShowExamineInfo(uid, false); // Hide SSD indicator
+
+            _transform.AttachToGridOrMap(projection);
             _mindSwap.Swap(uid, projection);
 
             _psionics.LogPowerUsed(uid, "telegnosis");
             args.Handled = true;
         }
         private void OnMindRemoved(EntityUid uid, TelegnosticProjectionComponent component, MindRemovedMessage args)
         {
+            // This is called during transfer to, so the MindSwappedComponent is still present.
+            if (TryComp<MindSwappedComponent>(uid, out var mindSwapped)){
+                _mind.ShowExamineInfo(mindSwapped.OriginalEntity, true);
+            }
             QueueDel(uid);
         }
+
+        public (EntityUid uid, TelegnosticProjectionComponent? comp) GetCasterProjection(EntityUid uid)
+        {
+            if (!TryComp<MindSwappedComponent>(uid, out var mindSwapped) ||
+                !TryComp<TelegnosticProjectionComponent>(mindSwapped.OriginalEntity, out var projection))
+            {
+                return (default, null);
+            }
+            return (mindSwapped.OriginalEntity, projection);
+        }
```

Comment:
change it to just return EntityUid? and check HasComp for the original entity since the component is never used